### PR TITLE
qa/cephfs: move run_shell to mount.py and add methods for negative testing

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Same as https://github.com/ceph/ceph/pull/33279, except for improvements. https://github.com/ceph/ceph/pull/33279 was reverted because it got merged even though tests for didn't pass.

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
Created new label "vstart_runner" so that both cephfs and dashboard teams can track changes in vstart_runner.py.

**Comment by rishabh-d-dave** [comment]
Ran client-recovery and volumes testsuite.
http://pulpito.ceph.com/rishabh-2020-04-17_10:27:46-fs-wip-rishabh-fs-move-run_shell-distro-basic-smithi/
http://pulpito.ceph.com/rishabh-2020-04-17_11:30:22-fs-wip-rishabh-fs-move-run_shell-distro-basic-smithi/

client-recovery passed fully and 2 out of 3 volumes job passed. The failure is unrelated IMHO for two reasons. First, actual tests hadn't begun to be executed, the job was still in "setup" phase and, second, the failure is not due to a call to run_shell().

Here's the traceback -
```
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 90, in run_tasks
    manager.__enter__()
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-fs-move-run_shell/qa/tasks/ceph.py", line 1539, in restart
    ctx.managers[cluster].wait_for_all_osds_up()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-fs-move-run_shell/qa/tasks/ceph_manager.py", line 2511, in wait_for_all_osds_up
    while not self.are_all_osds_up():
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-fs-move-run_shell/qa/tasks/ceph_manager.py", line 2501, in are_all_osds_up
    x = self.get_osd_dump()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-fs-move-run_shell/qa/tasks/ceph_manager.py", line 2303, in get_osd_dump
    return self.get_osd_dump_json()['osds']
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-fs-move-run_shell/qa/tasks/ceph_manager.py", line 2295, in get_osd_dump_json
    out = self.raw_cluster_cmd('osd', 'dump', '--format=json')
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-fs-move-run_shell/qa/tasks/ceph_manager.py", line 1363, in raw_cluster_cmd
    stdout=BytesIO(),
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 203, in run
    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 473, in run
    r.wait()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 162, in wait
    self._raise_for_status()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 184, in _raise_for_status
    node=self.hostname, label=self.label
CommandFailedError: Command failed on smithi119 with status 124: 'sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph --log-early osd dump --format=json'
```

@gregsfortytwo I realzie your opinion will most probably vary, let me know if you want me to retest this PR and/or you want me to run more testsuites.

**Comment by gregsfortytwo** [comment]
That failure looks like https://tracker.ceph.com/issues/45133 which popped up in our upgrade tests.

**Comment by rishabh-d-dave** [comment]
> That failure looks like https://tracker.ceph.com/issues/45133 which popped up in our upgrade tests.

Oh. Thanks for pointing that out. Copying the traceback from the [failing job's log](http://qa-proxy.ceph.com/teuthology/rishabh-2020-04-17_11:30:22-fs-wip-rishabh-fs-move-run_shell-distro-basic-smithi/4961909/teuthology.log) here for convenience -

```
2020-04-17T11:51:04.939 INFO:tasks.ceph.osd.1.smithi119.stderr: ceph version 16.0.0-831-gec1c2af (ec1c2af43f75f6c84efec35264da76fa0889bd43) pacific (dev)
2020-04-17T11:51:04.939 INFO:tasks.ceph.osd.1.smithi119.stderr: 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x154) [0x55d868d94814]
2020-04-17T11:51:04.939 INFO:tasks.ceph.osd.1.smithi119.stderr: 2: (ceph::__ceph_assertf_fail(char const*, char const*, int, char const*, char const*, ...)+0) [0x55d868d949ef]
2020-04-17T11:51:04.939 INFO:tasks.ceph.osd.1.smithi119.stderr: 3: (BlueStore::_prepare_ondisk_format_super(std::shared_ptr<KeyValueDB::TransactionImpl>&)+0x332) [0x55d8692cb3b2]
2020-04-17T11:51:04.940 INFO:tasks.ceph.osd.1.smithi119.stderr: 4: (BlueStore::_upgrade_super()+0x4fe) [0x55d8692cd2be]
2020-04-17T11:51:04.940 INFO:tasks.ceph.osd.1.smithi119.stderr: 5: (BlueStore::_mount(bool, bool)+0x599) [0x55d869320a29]
2020-04-17T11:51:04.940 INFO:tasks.ceph.osd.1.smithi119.stderr: 6: (OSD::init()+0x4d1) [0x55d868e31341]
2020-04-17T11:51:04.940 INFO:tasks.ceph.osd.1.smithi119.stderr: 7: (main()+0x3da5) [0x55d868d9bbe5]
2020-04-17T11:51:04.940 INFO:tasks.ceph.osd.1.smithi119.stderr: 8: (__libc_start_main()+0xe7) [0x7ef
```

**Comment by gregsfortytwo** [comment]
http://pulpito.front.sepia.ceph.com/gregf-2020-04-21_21:12:13-fs-wip-greg-testing-421-1-distro-basic-smithi/ looks good; few infra bugs and known issues

File: qa/tasks/cephfs/mount.py [hunk 1]
```diff
@@ -194,15 +194,89 @@ def run_python(self, pyscript, py_version='python3'):
...
+        if isinstance(args, str):
+            args = 'sudo ' + args
+        if isinstance(args, list):
+            args.insert(0, 'sudo')
+        if not cwd:
+            cwd = self.mountpoint
+
+        return self.client_remote.run(args=args, wait=wait, stdin=stdin,
+                                      stdout=BytesIO(), stderr=BytesIO(),
+                                      check_status=check_status, cwd=cwd)
```
Comment (jtlayton) [inline qa/tasks/cephfs/mount.py:239]: nit: this could just be a wrapper around run_as_user. `sudo -u root` does work.

**Comment by gregsfortytwo** [comment]
Thanks for the review @jtlayton

File: qa/tasks/cephfs/mount.py (hunk 1 above)
Comment (rishabh-d-dave) [reply qa/tasks/cephfs/mount.py:239]: I'll get that done on a separate PR.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/mount.py:239]: https://github.com/ceph/ceph/pull/35540

**Pull request merged**